### PR TITLE
Radical Red Save Compatibility

### PR DESCRIPTION
**Pull request by evanwporter**
See pokemon-file-js PR

### CODE REVIEW

**Comment by andrewbenington** [comment]
It might take me a few days to look at these PRs, but I will do so and give feedback if necessary. Looking forward to getting this working!

File: tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "allowImportingTsExtensions": true,
     "resolveJsonModule": true,
     "isolatedModules": true,
+    "noUnusedParameters": false,
```
Comment (andrewbenington) [inline tsconfig.json]: Can we keep this on? There shouldn't be a reason to have a function with an unused parameter.

File: tsconfig.node.json [hunk 2]
```diff
@@ -4,5 +4,6 @@
   "compilerOptions": {
     "types": ["electron-vite/node"],
     "target": "ESNext"
-  }
+  },
+  "noUnusedParameters": false,
```
Comment (andrewbenington) [inline tsconfig.node.json]: Same here

File: src/types/types.ts [hunk 3]
```diff
@@ -219,7 +222,7 @@ export type Move = {
   generation: string
   power?: number
   pp: number
-  pastGenPP: {
+  pastGenPP?: {
```
Comment (andrewbenington) [inline src/types/types.ts]: If you make this optional, update the lines in `getMoveMaxPP()` in `src/types/pkm/util.ts`

File: src/consts/RRTransferMons.ts [hunk 4]
```diff
@@ -0,0 +1,1015 @@
...
+    1017,
+    1018,
+    1019,
+    1020,
+    1021,
+    1022,
+    1023,
+    1024,
+    1025,
+]
```
Comment (andrewbenington) [inline src/consts/RRTransferMons.ts]: nitpick, but add a newline at the end of this file

**Review by andrewbenington — DISMISSED** [review]
Nice, just a couple fixes needed

**Comment by andrewbenington** [comment]
Also, there are merge conflicts that need to be resolved

File: tsconfig.json (hunk 1 above)
Comment (evanwporter) [reply tsconfig.json]: Yeah for sure I thought I removed changes to this file from the commit...

File: src/consts/RRTransferMons.ts (hunk 4 above)
Comment (evanwporter) [reply src/consts/RRTransferMons.ts]: For sure.

**Comment by evanwporter** [comment]
I think the merge conflicts need to be resolved by you. But they are small so you can just keep the original files then make the changes manually to OHPKM and util

File: package-lock.json
Comment (evanwporter) [inline package-lock.json:1]: Honestly just ignore this entire file. It gets regenerated anyways.

File: src/consts/RRTransferMons.ts
Comment (evanwporter) [inline src/consts/RRTransferMons.ts:1]: This file was created so `TransferRestrictions.ts` was not junked up. I automatically generated it.

File: src/types/SAVTypes/G3RRSAV.ts [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+          mon &&
+          mon.trainerID === this.tid &&
+          mon.secretID === this.sid &&
+          mon.trainerName === this.name
+        ) {
+          this.origin = mon.gameOfOrigin
+        }
+      })
+    })
+    const filePathElements = splitPath(this.filePath)
```
Comment (evanwporter) [inline src/types/SAVTypes/G3RRSAV.ts]: This part needs to be changed to just `GameOfOrigin.FireRed` or `GameOfOrigin.RadicalRed` at some point in the future. I haven't looked too deeply into how the `origin` property is used which is why I haven't modified it.

File: src/types/pkm/OHPKM.ts [hunk 6]
```diff
@@ -86,7 +86,7 @@ export class OHPKM {
 
   bytes = new Uint8Array(433)
 
-  constructor(arg: PKM | OHPKM | Uint8Array<ArrayBuffer>) {
+  constructor(arg: PKM | OHPKM | Uint8Array) {
```
Comment (evanwporter) [inline src/types/pkm/OHPKM.ts:89]: VSCode/TypeScript compiler was giving me an error here, but if this works for you then you can remove this edit.

**Comment by evanwporter** [comment]
Yeah this is really hype if we can get this working. I work on a windows and I tried building the repo but I got a bunch of error :/ I fixed the errors resulting from

**Comment by andrewbenington** [comment]
I've worked with FOSS repos before but never had my own, so this is a learning process for me too. I'm glad you are also excited about the project, and if you'd like I can give you permissions to directly contribute to this repo instead of needing your own fork. I can look into that after work today.

And yeah I ran into some issues building on a Windows VM and had to remove some typing errors with `Uint8Array`s, which it looks like you fixed on this independently, so thank you. What other build errors do you have? I can build the main branch on Windows just fine now, but if you're still running into issues I want to make sure there isn't anything I'm missing. Can you make a [discussion post](https://github.com/andrewbenington/OpenHome/discussions) with errors and/or screenshots of your issues?

File: src/types/pkm/OHPKM.ts (hunk 6 above)
Comment (andrewbenington) [reply src/types/pkm/OHPKM.ts:89]: Yep I fixed this on the main branch, thanks for catching it as well

**Pull request merged**